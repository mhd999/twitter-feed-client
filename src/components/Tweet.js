// @flow

import React from 'react';
import styler from 'react-styling';

const style = styler `
    Tweet
        background: gray
        margin: 5px
`;
const Tweet = (props: any) => {
    const tweet = props.data;
    return (
        <li key={tweet.id} style={style.Tweet}>
            <span> {tweet.text} </span>
        </li>
    );
}

export default Tweet;