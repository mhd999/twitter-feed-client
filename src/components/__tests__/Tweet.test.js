import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Tweet from '../Tweet';

const tweets ={"id":808422013928935400,"text":"RT @wwf_uk: 70% of Norway’s wolf population could be hunted &amp; killed this winter. Help @WWFNorge #SaveOurWolves:… "};

describe('<Tweet />', () => {
  it('renders children', () => {
    const component = shallow(
      <Tweet data={tweets} />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
});
