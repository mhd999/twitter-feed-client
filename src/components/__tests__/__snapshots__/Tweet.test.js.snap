exports[`<Tweet /> renders children 1`] = `
<li
  style={
    Object {
      "background": "gray",
      "margin": "5px",
    }
  }>
  <span>
     
    RT @wwf_uk: 70% of Norway’s wolf population could be hunted &amp; killed this winter. Help @WWFNorge #SaveOurWolves:… 
     
  </span>
</li>
`;
